snapcraft.yaml

```diff
@@ -93,21 +93,52 @@ parts:
     plugin: nil
     build-packages:
       - wget
-    override-pull: |
-      # Do not use rustup to work around https://forum.snapcraft.io/t/armhf-builds-on-launchpad-timing-out/31008
+    override-pull: wget -O rustup.sh https://sh.rustup.rs
+    override-build: |
```

Comment:
Why this change? Reading the snapcraft forum it looks like the original builder issue was fixed and it might make sense to remove the workaround but I'm unsure what it has to do with the debug symbols? In fact the firefox snap still has that same code and working debug symbol.

If it's unrelated can we get it split in another PR (and we will want to do the same for firefox and try to keep those packages closer if we can)

Reply:
I took that from the Firefox snapcraft: https://github.com/canonical/firefox-snap/blob/nightly/snapcraft.yaml#L139

Reply:
Hum, seems firefox is also inconsistent between channels :-/ 

But checking the nightly firefox build those seem to fail on armhf

https://launchpadlibrarian.net/775641855/buildlog_snap_ubuntu_noble_armhf_firefox-snap-nightly_BUILDING.txt.gz

```
[10/Feb/2025:01:47:52 +0000] "CONNECT sh.rustup.rs:443 HTTP/1.1" 200 31691 "-" "Wget/1.21.4"
:: Connecting to 10.10.10.1:8222... connected.
:: Proxy request sent, awaiting response... 404 Not Found
```

Unsure if that's a sign we should have the workaround there also instead?

In any case that seems orthogonal to the debug symbol so can we remove that part of the change and keep it for another PR instead?

Reply:
Then we should fix firefox Snap

Reply:
So yeah I was not completely dumb, I changed that because the build of `dump-syms` was failing:
```
2025-02-10 14:27:04.748 :: 2025-02-10 14:27:03.174 :: + /root/.cargo/bin/cargo build --release
2025-02-10 14:27:04.748 :: 2025-02-10 14:27:03.174 :: /tmp/tmpc6r81ibg/scriptlet.sh: line 7: /root/.cargo/bin/cargo: No such file or directory
2025-02-10 14:27:04.748 :: 2025-02-10 14:27:03.175 'override-build' in part 'dump-syms' failed with code 127.
2025-02-10 14:27:04.748 :: 2025-02-10 14:27:03.176 Detailed information:
2025-02-10 14:27:04.748 :: :: + /root/.cargo/bin/cargo build --release
2025-02-10 14:27:04.748 :: :: /tmp/tmpc6r81ibg/scriptlet.sh: line 7: /root/.cargo/bin/cargo: No such file or directory

Reply:
(because there's a cargo already installed)